pcsx2/gui/FrameForGS.cpp

```diff
@@ -438,7 +438,7 @@ GSFrame::~GSFrame() throw()
 void GSFrame::OnCloseWindow(wxCloseEvent& evt)
 {
 	sApp.OnGsFrameClosed( GetId() );
-	Hide();		// and don't close it.
+	CoreThread.Reset(); // reset thread to avoid suspend status on close event.
```

Comment:
Suspending alone will not fix the memory card dialog issue since suspend still means ```CoreThread.IsRunning()``` will return true and eventually ```CoreThread.Resume();``` will be done and the closed GS frame will be back on, Reset is the only solution I've managed to find. Reset can be considered as the same function as shutdown so I don't believe it could cause any issues but it's probable though.